airflow/sensors/sql_sensor.py

```diff
@@ -34,22 +34,25 @@ class SqlSensor(BaseSensorOperator):
     :param sql: The sql to run. To pass, it needs to return at least one cell
         that contains a non-zero / empty string value.
     :type sql: str
+    :param parameters: The parameters to render the SQL query with (optional).
+    :type parameters: mapping or iterable
     """
     template_fields = ('sql',)
     template_ext = ('.hql', '.sql',)
     ui_color = '#7c7287'
 
     @apply_defaults
-    def __init__(self, conn_id, sql, *args, **kwargs):
-        self.sql = sql
+    def __init__(self, conn_id, sql, parameters=None, *args, **kwargs):
         self.conn_id = conn_id
+        self.sql = sql
+        self.parameters = parameters
         super(SqlSensor, self).__init__(*args, **kwargs)
 
     def poke(self, context):
         hook = BaseHook.get_connection(self.conn_id).get_hook()
 
-        self.log.info('Poking: %s', self.sql)
-        records = hook.get_records(self.sql)
+        self.log.info('Poking: %s (with parameters %s)', self.sql, self.parameters)
+        records = hook.get_records(self.sql, self.parameters)
```

Comment:
Do all the sql type hooks(druid, mysql, postgres etc) are all inherited from dbapi hook? If that's case, the change LGTM

Reply:
If not, we should change that hook to inherit from dbapi hook.

Reply:
Yes, hooks of druid, mysql, mssql, Postgres, oracle etc are all inherited from dbapi hook.

For `SqlSensor`, the `get_hook()` in `BaseHook.get_connection(self.conn_id).get_hook()` will decide which exact hook to use based on the connection type (reference: https://github.com/apache/airflow/blob/master/airflow/models/connection.py#L184).

Reply:
But I check`CloudSqlDatabaseHook `(https://github.com/apache/airflow/blob/6b38649fa6cdf16055c7f5458050c70f39cac8fd/airflow/contrib/hooks/gcp_sql_hook.py#L669) which doesn't seem to inherit from dbapi hook. 

But the I assume the sqlSensor could use CloudSqlDatabasehook? Given there are many connection hooks defined, a safer approach would be check if the hook is an instance of dbapi hook, use the `records = hook.get_records(self.sql, self.parameters)`, otherwise fall back to `records = hook.get_records(self.sql)`. what you think?

Reply:
Actually `SqlSensor` can not support `CloudSqlDatabaseHook`. 

Eventually `SqlSensor` uses `get_records` method to retrieve the records from DB, while `get_records` method is not available in `CloudSqlDatabaseHook` at all.

Reply:
I see, but would a check on hook type be safer?

Reply:
Get what you mean.

Given the limitation in https://github.com/apache/airflow/blob/master/airflow/models/connection.py#L184 and the implementation of each hook, only the connection types below are supported by `SqlSensor`:

- 'google_cloud_platform'
- 'jdbc'
- 'mssql'
- 'mysql'
- 'oracle'
- 'postgres'
- 'presto'
- 'sqlite'
- 'vertica'

I will add a check.